salt/modules/cmdmod.py

```diff
@@ -625,7 +626,15 @@ def _get_stripped(cmd):
             if err is not None:
                 err = err.rstrip()
         ret['pid'] = proc.process.pid
-        ret['retcode'] = proc.process.returncode
+        _returncode = proc.process.returncode
+        ret['retcode'] = _returncode
+        if success_exit_codes:
+            if not isinstance(success_exit_codes, list):
+                success_exit_codes = list(success_exit_codes)
+            # Ensure all our retcodes are ints
+            success_exit_codes = [int(i) for i in success_exit_codes]
+            if _returncode in success_exit_codes:
+                ret['retcode'] = 0
```

Comment:
In many places in salt we support both a Python and comma-separated list. We have a helper function for this, `salt.utils.args.split_input()`, which would be a good fit here.

Also , I noticed that there is a similar block below. How about moving the evaluation of `success_exit_codes` up higher in the function and doing something like this:

```python
if success_exit_codes is None:
    success_exit_codes = [0]
else:
    try:
        success_exit_codes = [int(i) for i in salt.utils.args.split_input(success_exit_codes)]
    except ValueError:
        raise SaltInvocationError(
            'success_exit_codes must be a list of integers'
        )
```

Then you could just do a simple membership check:

```python
if ret['retcode'] in success_exit_codes:
```

Reply:
I'm also wondering if maybe this should be called `success_retcodes` instead of `success_exit_codes`. We already use `retcode` everywhere else in cmdmod.py to refer to the exit code of a command, so it may make sense to rename this for clarity/consistency.